go/kbfs/libhttpserver/content_type_utils.go

```diff
@@ -88,20 +94,91 @@ func (w *contentTypeOverridingResponseWriter) override() {
 	w.original.Header().Set("X-Content-Type-Options", "nosniff")
 }
 
+func (w *contentTypeOverridingResponseWriter) checkViewTypeInvariance() error {
+	if len(w.viewTypeInvariance) == 0 {
+		return nil
+	}
+	if _, _, i := GetGUIFileContext(w.original.Header()); i == w.viewTypeInvariance {
+		return nil
+	}
+	w.original.WriteHeader(http.StatusPreconditionFailed)
+	return errors.New("viewTypeInvariance doesn't match")
+}
+
 func (w *contentTypeOverridingResponseWriter) Header() http.Header {
 	return w.original.Header()
 }
 
 func (w *contentTypeOverridingResponseWriter) WriteHeader(statusCode int) {
 	w.override()
+	if err := w.checkViewTypeInvariance(); err != nil {
+		return
+	}
 	w.original.WriteHeader(statusCode)
 }
 
 func (w *contentTypeOverridingResponseWriter) Write(data []byte) (int, error) {
 	w.override()
+	if err := w.checkViewTypeInvariance(); err != nil {
+		return 0, err
+	}
 	return w.original.Write(data)
 }
 
+func beforeSemicolon(str string) string {
+	semicolonIndex := strings.Index(str, ";")
+	if semicolonIndex > 0 {
+		str = str[:semicolonIndex]
+	}
+	return strings.ToLower(strings.TrimSpace(str))
+}
+
+var supportedImgMimeTypes = map[string]bool{
+	"image/png":  true,
+	"image/jpeg": true,
+	"image/gif":  true,
+	"image/webp": true,
+}
+
+func getGetGUIFileContextReturnType(
+	viewType keybase1.GUIViewType, contentType string) (
+	keybase1.GUIViewType, string, string) {
+	return viewType, contentType, strconv.Itoa(int(keybase1.GUIViewType_DEFAULT))
```

Comment:
Also, the `strconv` result can be saved into a constant (even if it's technically a `var`) to avoid re-doing it on every request.